server/lib/controller/user_controller.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:aqueduct/aqueduct.dart';
+
+import '../proto/create_user_request.pb.dart';
+import '../service/firestore.dart';
+import '../utils.dart';
+
+/// A controller that handles operations on users.
+class UserController extends ResourceController {
+  /// The client used to interact with Firestore.
+  final FirestoreClient _client;
+
+  /// The AES key used to encrypt/decrypt the refresh token.
+  final String _aesKey;
+
+  /// Creates an instance of [UserController].
+  UserController(this._client, this._aesKey);
+
+  /// The accepted content types for this controller.
+  @override
+  List<ContentType> acceptedContentTypes = [
+    ContentType('application', 'x-protobuf')
+  ];
+
+  /// Creates a user with a user ID and encrypted refresh token.
+  ///
+  /// Throws an [ArgumentError] if the request does not contain an Authorization
+  /// header. Throws an [ApiRequestError] if the Firestore API returns an error.
+  /// Returns a 200 OK [Response] upon success.
+  @Operation.post()
+  Future<Response> createUser(@Bind.body() List<int> body) async {
```

Comment:
nit:  In general the create() method needs to return the resources created in stead of empty() following the restful design . I assume this function will be called in the background of the server, so it is not customer initiated and visible operation. Also seems that directly binding the body with a message like User has better readability. Maybe we can add a TODO here, and refactor later.

Reply:
Ok, that makes sense